crates/wasmtime/src/runtime/vm/instance/allocator/wasmfx_allocator.rs

```diff
@@ -0,0 +1,261 @@
+// WasmFX fiber stack allocators
+//
+// * on_demand: allocates memory lazily
+// * pooling: preallocates a chunk of memory eagerly
+//
+
+use anyhow::Result;
+use wasmtime_continuations::WasmFXConfig;
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "wasmfx_baseline")] {
+        use wasmtime_fiber as fiber_impl;
+    } else {
+        use crate::vm::fibre as fiber_impl;
+    }
+}
+
+pub type FiberStack = fiber_impl::FiberStack;
+
+#[cfg(not(feature = "wasmfx_pooling_allocator"))]
+pub mod wasmfx_on_demand {
+    use super::*;
+
+    #[derive(Debug)]
+    pub struct InnerAllocator {
+        stack_size: usize,
+    }
+
+    impl InnerAllocator {
+        pub fn new(config: &WasmFXConfig) -> Result<Self> {
+            Ok(InnerAllocator {
+                stack_size: config.stack_size,
+            })
+        }
+
+        pub fn allocate(&self) -> Result<FiberStack> {
+            cfg_if::cfg_if! {
+                if #[cfg(all(feature = "unsafe_wasmfx_stacks", not(feature = "wasmfx_baseline")))] {
+                    Ok(fiber_impl::FiberStack::malloc(self.stack_size).ok_or_err(|_| anyhow::anyhow!("Fiber stack allocation failed"))?)
+                } else {
+                    Ok(fiber_impl::FiberStack::new(self.stack_size).map_err(|_| anyhow::anyhow!("Fiber stack allocation failed"))?)
+                }
+            }
+        }
+
+        pub fn deallocate(&self, _stack: &FiberStack) {
+            // The on-demand allocator has no further bookkeeping for fiber stacks
+        }
+    }
+}
+
+#[cfg(feature = "wasmfx_pooling_allocator")]
+pub mod wasmfx_pooling {
+    use super::*;
+
+    use crate::runtime::vm::instance::allocator::pooling::{
+        index_allocator::{SimpleIndexAllocator, SlotId},
+        round_up_to_pow2,
+    };
+    use crate::runtime::vm::sys::vm::{commit_pages, decommit_pages};
+    use crate::vm::Mmap;
+    use anyhow::{anyhow, bail, Context, Result};
+
+    /// Represents a pool of execution stacks (used for the async fiber implementation).
```

Comment:
Used for the wasmfx implementation! :) The sentence below should probably be adapted to reflect our usage of stacks more concretely.

Reply:
Yes.